File: backend/internal/domain/child.go
package domain

import "time"

type Child struct {
	ID        string    `json:"id" firestore:"-"`
	FamilyID  string    `json:"familyId" firestore:"-"` // Parent document reference
	Name      string    `json:"name" firestore:"name"`
	BirthDate time.Time `json:"birthDate" firestore:"birthDate"`
	PhotoURL  string    `json:"photoURL,omitempty" firestore:"photoURL,omitempty"`
	CreatedAt time.Time `json:"createdAt" firestore:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt" firestore:"updatedAt"`
}

File: app/lib/features/family/domain/entities/child.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'child.freezed.dart';
part 'child.g.dart';

@freezed
class Child with _$Child {
  const factory Child({
    required String id,
    required String familyId,
    required String name,
    required DateTime birthDate,
    String? photoURL,
    required DateTime createdAt,
    required DateTime updatedAt,
  }) = _Child;

  factory Child.fromJson(Map<String, dynamic> json) => _$ChildFromJson(json);
}
